### DISCUSSION TITLE
Power readings are way higher than reality

### DISCUSSION THREAD

**Discussion by catatung** [emporia-vue-local/esphome#117] [Q&A]
Hey everybody--

I recently flashed a vue2 with the ESP32 firmware and have added it to a homeassistant instance (running HAOS on a proxmox VM if it matters) -- I was able to pull in the entities however, the values are incorrect.

The software pulled in POWER and DAILY ENERGY for each circuit.  However, after comparing with a standalone current clamp meter, I found a huge disparity in some.  Two examples are below:

Circuit 6 current clamp meter reading: 12.10A
HA reports 2670W
2670W / 117.6V = 22.70A

Circuit 12 current clamp meter reading: 0.750A
HA reports 98.9W
98.9W / 117.6 = 0.84A

One is way off, the other is close.  I switched the current clamp jacks and got the same readings, so I have ruled out issue of a bad CT sensor.  In searching I read about assigning the correct phase for each circuit in configuration, but both phases read 117.6v, so I didn't bother to get to that just yet, since it wouldn't be the problem here as both are identical.

Does anybody have any ideas what I could do to fix this or tests I could perform to nail down what the cause of these inaccuracies are?  I've read that the native reporting of current is off, and that it should be mathematically computed within HA to get the correct data, but before I even get to that, I'm seeing the wrong values.  Is there some other data available via JSON/esp32home that HA did not pull in that I should be using to calculate the wattage/current draw?

Thanks for any help
Randy

**Comment by flaviut** [comment]
> both phases read 117.6v, so I didn't bother to get to that just yet, since it wouldn't be the problem here as both are identical

The RMS voltage may be identical, but the phases [are definitely not](http://hyperphysics.phy-astr.gsu.edu/hbase/electric/phase3.html)! Current & voltage is measured by the vue thousands of times a second, while your multimeter only measures a long-term average.

Please update your & come back if you're still have trouble.

**Reply by catatung** [reply]
Hi Flaviut - Thanks for the quick response.  What would you recommend I do next?  I imagine there could be a disparity between RMS voltage and the long term average as shown by a meter, but I would not think enough to cause a reading 100% higher than actual.  I trust the current-clamp meter is close to accurate more than I do the VUE2 at this point.  Also this is only a 15 amp circuit, so I am pretty sure 22.7 amps is not what's actually present on the wire.  

I have not connected the A&B clamps yet on the two phases - is it possible the absence of those could cause this issue?  I would think not, but of course I could be wrong!

**Reply by catatung** [reply]
*Also, that value of 117.6 I took from the dashboard in homeassistant.  I used the standalone meter for current measurement with it's current clamp, for comparison to the VUE2/HA.

**Reply by flaviut** [reply]
Sorry, I threw a bunch of big words at you! The multimeter voltage is both the RMS & long-term average voltage.

> I have not connected the A&B clamps yet on the two phases - is it possible the absence of those could cause this issue?

Nope, that's fine.

> What would you recommend I do next?

- Double check that the proper phase is set on every channel in your config.
- Double check that your current clamp's jack is fully inserted. I found these to be pretty annoying during install, I had jacks come partially out several times.

**Reply by catatung** [reply]
Alright I can do a little digging into the config files and see about the phase association.  I was avoiding spending time to do that because homeassistant is showing 117.6v for both phases, so I thought it would get calculated just the same even if the phase was set wrong since apparently the vue2/HA has interpreted both phases as equal.  I do see it bounce around .1 or so from time to time, so it is a live measurement and not a stuck, stale value.

I will look at the config file and confirm the phase associations to the circuits.  Thanks for your advice and comments.

Randy

**Reply by catatung** [reply]
And yes I have read about the jacks that do not seat properly, seems like a design flaw as many people have complained about that!  I've checked that they are in good and tight, or as tight as can be.  Once I have everything working, I may 3d print a long bar with a couple of slotted openings for screws and slide it over to keep pressure on all of the jacks and then tighten down the screws to make sure they don't move.  The unit itself is mounted to a wooden board already with a 3d printed bracket, so it is stable and secure.

**Reply by catatung** [reply]
I just retrieved my configuration via settings --> integrations --> emporiavue2 ESPHome --> Device Info --> Download Diagnostics.   I do not see a phase identification on any of the entries.  Circuit 6 reads, for example:

`"object id": "circuit_6_power",
"key": 1400347427,
"name": "Circuit 6 Power",
"unique_id": "emporiavue2sensorcircuit_6_power",
"disabled_by_default": false,
"icon": "",
"entity_category": 0,
"device_class": "power",
"unit of measurement": "W",
"accuracy_decimals": 1,
"force_update": false,
"state_class": 1,
"last_reset_type": 0`

The only mention of phase in this config are the object id's for "phase_a_power", "phase_b_power", "phase_a_voltage", and "phase_b_voltage."

Am I looking in the right place?  Should there be a phase assignment in these object id's?

Thanks for your help
Randy

**Reply by flaviut** [reply]
The config is the file that you copy-pasted and starts with

```
esphome:
  name: emporiavue2

external_components:
  - source: github://emporia-vue-local/esphome@dev
    components: [ emporia_vue ]

esp32:
  board: esp32dev
  framework:
    type: esp-idf
    version: recommended

# Enable Home Assistant API
api: {"password": "<ota password>"}
ota: {"password": "<ota password>"}

# Enable logging
logger:

wifi:
  ssid: "<wifi ssid>"
  password: "<wifi password>"

preferences:
  # the default of 1min is far too short--flash chip is rated
  # for approx 100k writes.
  flash_write_interval: "48h"
```

Example phase line:

```
      - { phase_id: phase_a, input:  "1", power: { name:  "Circuit 1 Power", id:  cir1, filters: [ *moving_avg, *pos ] } }
```

**Reply by catatung** [reply]
Hi Flaviut;

Thanks for your example -- I reviewed my files from when I produced the firmware and flashed the vue2 unit with ESPHome-Flasher (vue2.yaml) and I see phase information for every circuit, including the exact line you copied.  However, when I go to settings --> integrations --> emporiavue2 ESPHome --> Device Info --> Download Diagnostics on the HomeAssistant, I only get the data which I previously pasted to you, which lacks any indication of phase information.

Is there somewhere else I should be looking in the homeassistant interface to add this phase information?  It seems to me it should already be there if it was in the config file when I flashed the unit with ESPHome-Flasher.

Thanks for your time
Randy

**Reply by flaviut** [reply]
No clue what "Device Info --> Download Diagnostics" does, but all changes need to be done in the file you found, and the firmware re-compiled and re-flashed each time each time the config is changed.

The firmware flashing can be done OTA [through the command line](https://esphome.io/guides/cli.html#upload-command) or [ESPHome dashboard](https://esphome.io/guides/getting_started_hassio.html#dashboard-interface).

**Reply by catatung** [reply]
Hi Flaviut,

That file I found/showed to you was from the download diagnostics link.  It sent a text file to my browser.  Image is attached so you can see them.

Since this is a HAOS installation, I do not have the linux command line shell which I am most familiar with for finding and editing configuration files.  When I originally flashed the unit, I used a vue2.yaml file to generate the firmware which had numerous references to phase association for all of the circuits.  Is there a different place you recommend I can look to check if the phase information is set correctly?  It appears either the file "download diagnostics" sent me is incomplete, or the initial flashing process paid no regard to the phase information in the vue2.yaml file.

Otherwise I guess, if there are no other ideas, I could try flashing again.  Though I would still like to find out how I can confirm from the HA/ESP32Home interface if the phase information ported over as expected.

Thanks for any ideas or suggestions
Randy


![ha-dd](https://user-images.githubusercontent.com/40948885/218263127-218e1c28-221f-494b-8727-f96165b09e0d.JPG)

**Reply by flaviut** [reply]
>  Is there a different place you recommend I can look to check if the phase information is set

There is no way to do this. [This is the closest you can get](https://github.com/emporia-vue-local/esphome/discussions/111#discussioncomment-4836412).

The only way to confirm is to reflash the config.

Once you've checked the phases in the config match the phases in your panel, and you've reflashed the Vue, please let me know if the data still looks wrong. Also make sure that your multiply values in `filters: [ *moving_avg, *pos, multiply: 2 ]` are correct. [Step 9 in the Vue manual has instructions](https://uploads-ssl.webflow.com/5fff2b7694451e66ba2f5a3d/628fdc1da2b2197b6ca9b8e8_Gen%202%20Emporia%20Vue%20Installation%20Guide.pdf).

**Reply by catatung** [reply]
Hi Flaviut,

I took myself back to basics today and took some time to RTFM -- and figured out what I had done wrong, my mistake was staring at me the whole time!

I had made a temporary plug for the unit so I could power it up at my computer desk and flash the firmware.  After examining the documentation, I joined the blue and white wires for connection to neutral, and joined the red and black wires for connection to hot. I wired these to an electrical outlet plug, and plugged it into my Variac (for some added isolation via the transformer windings)

Once I was done flashing the firmware, I brought it down to the panel and out of convenience just plugged in the cable I had created at the desk, forgetting that the two phases need to be connected separately inside the panel!

I am going to assume this is my problem -- I will get some time soon to disconnect my plug connection and attach it correctly inside the panel, taking note of the phases on each breaker.  I will then update my .yaml file and create the .bin firmware file, and reflash the unit.  I'll let you know how everything is once I've done that.  I am thinking this will solve my problems.

Thanks for pushing me in the direction of the phase management and helping me to realize my mistake! :)

Randy

**Reply by catatung** [reply]
I've wired the 4 pin connector correctly now to my panel -- I now have different readings for Phase A and Phase B voltage, as you previously indicated would absolutely be the case.

I've mapped out which circuits are on which phases, and before looking into modifying my configuration to reflash the unit, I reviewed my previous YAML file which I used to create the firmware BIN file which was flashed to the unit.  I found this:

```
- { phase_id: phase_a, input:  "6", power: { name:  "Circuit 6 Power", id:  cir6, filters: [ *moving_avg, *pos, multiply: 2 ] } }
- { phase_id: phase_a, input: "12", power: { name: "Circuit 12 Power", id: cir12, filters: [ *moving_avg, *pos, multiply: 2 ] } }

```

I believe this is my problem - the multiplier is set to 2 on the two specific circuits that I am seeing the implausible current draw.  I believe this was a starting point configuration file from following instructions when I initially flashed the unit, however I didn't take any time to review the configuration file, because I read that future changes could be easily made and not require grounding the GPIO to flash the unit.

I went to update my config file, starting with validating the .YAML file which I know I used previously.  I received this message when running "esphome ./vue2.yaml config"

```
INFO Reading configuration ./vue2.yaml...
Failed config

esphome: None

  'platform' is a required option for [esphome].
  name: emporiavue2


```
I know I used this file previously when I flashed via the windows utility.  Now that it is on my network, I am on Linux, and receiving this message.  Do you have any idea why?  I'm considering pulling the unit back and grounding the GPIO and flashing it the way I initially did, but doing it from Linux would be so much easier.

As always thanks for any help!
Randy

**Reply by flaviut** [reply]
Maybe try updating your esphome version--I just updated emporia-vue-local/esphome to the febuary edition a few hours ago. If that doesn't work, try clearing caches and so on.

The issue sounds like it's in the configuration step, which comes before the compilation & the flashing steps.
